diff --git a/c++/inchworm/solver_core.cpp b/c++/inchworm/solver_core.cpp
index f491117..e7275c4 100644
--- a/c++/inchworm/solver_core.cpp
+++ b/c++/inchworm/solver_core.cpp
@@ -78,9 +78,9 @@ namespace inchworm {

     if (sp.partition_method == "automatic") {
       ASSERT(sp.quantum_numbers.empty());
-      ad_imp = {sp.h_imp + h_hyb_ED, fops, create_effective_hyb(constr_params.gf_struct)};
+      ad_imp = atom_diag{sp.h_imp + h_hyb_ED, fops, create_effective_hyb(constr_params.gf_struct)};
     } else if (sp.partition_method == "quantum_numbers") {
-      ad_imp = {sp.h_imp + h_hyb_ED, fops, sp.quantum_numbers};
+      ad_imp = atom_diag{sp.h_imp + h_hyb_ED, fops, sp.quantum_numbers};
     } else {
       TRIQS_RUNTIME_ERROR
          << "Unknown partition method! Please choose 'automatic' or 'quantum_number' and set solve_params.quantum_numbers accordingly";
